fault/verilator_target.py

```diff
@@ -46,9 +52,20 @@ def __init__(self, circuit, actions, directory="build/",
         self.include_verilog_files = include_verilog_files
         self.magma_output = magma_output
 
+    @staticmethod
+    def make_array_action_code(i, action):
```

Comment:
nit: `generate` rather than `make`

Reply:
Handy function! But it seems it only handles one level of nesting. Perhaps we could throw the multi-level flattening into `flatten`? Or explicitly error if we don't want to support it (though I don't see any reason it wouldn't work here).

Reply:
it handles multiple levels of nesting, because it recursively calls `generate_action_code` which, if it encounters a nested array, will return a flattened list, so our `flatten` only needs to handle one level of flattening (assuming that everything is recursively flattened).

Reply:
e439fa3 tests this explicitly

Reply:
Ah, right, good point.